### PR TITLE
feat(ci): Add license workflow

### PR DESCRIPTION
**Pull request by sakoush**
**What this PR does / why we need it**:

This PR regenerates the 3rd party licenses and creates a PR with the change if required.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by sakoush** [comment]
tested in my  fork: https://github.com/sakoush/seldon-core/actions/runs/8363864806/job/22903578509

File: .github/workflows/license.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: V2 Licenses
+
+on:
+  schedule:
+    # * is a special character in YAML so you have to quote this string
+    - cron: "23 18 * * *"
```
Comment (RafalSkolasinski) [inline .github/workflows/license.yml:6]: Why "At 06:23 PM"? :)

File: .github/workflows/license.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  workflow_dispatch:
+
+jobs:
+  licenses:
+    runs-on: ubuntu-latest
+    if: github.repository == 'SeldonIO/seldon-core' # Do not run this on forks.
+    steps:
+      - name: Get github access token
+        run: |
+          echo $API_KEY > ~/.github_api_token
```
Comment (RafalSkolasinski) [inline .github/workflows/license.yml:16]: shouldn't it be sth along 
```
${{ secrets.GH_API_TOKEN }}
```
?

File: .github/workflows/license.yml (hunk 1 above)
Comment (sakoush) [reply .github/workflows/license.yml:6]: was following MLServer: https://github.com/SeldonIO/MLServer/blob/0b88ef73b6066663a4b074c20f9bb3a4c1dac0a2/.github/workflows/licenses.yml#L6 

but yeah I will change to perhaps midnight

**Review by RafalSkolasinski — APPROVED** [review]
LGTM

File: .github/workflows/license.yml (hunk 1 above)
Comment (sakoush) [reply .github/workflows/license.yml]: done in [deb2c9f](https://github.com/SeldonIO/seldon-core/pull/5465/commits/deb2c9fd30b5a99a0b36aa764387e21d3ea8b944)

**Pull request merged**